sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala

```diff
@@ -264,9 +265,16 @@ trait ProgressReporter extends Logging {
     if (lastExecution == null) return Nil
     // lastExecution could belong to one of the previous triggers if `!hasExecuted`.
     // Walking the plan again should be inexpensive.
+
+    val shufflePartitionValue = sparkSession.conf.getOption(SHUFFLE_PARTITIONS.key).getOrElse("-1")
+    val numShufflePartitions: Long = try {
+      shufflePartitionValue.toLong
+    } catch {
+      case e: NumberFormatException => -1L
+    }
     lastExecution.executedPlan.collect {
       case p if p.isInstanceOf[StateStoreWriter] =>
-        val progress = p.asInstanceOf[StateStoreWriter].getProgress()
+        val progress = p.asInstanceOf[StateStoreWriter].getProgress(numShufflePartitions)
```

Comment:
Shall we read the value from the physical plan? StateStoreWriter has a method `stateInfo` which you can find a number of shuffle partitions. That value is closer to the reality, as we build a child distribution requirement based on that value.

https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StatefulOperatorPartitioning.scala

This is also future-proof - if we want to set the different shuffle partitions per operator, we will set the value differently for stateInfo.

Reply:
Let me give it a try.